### PR TITLE
Dexcom changes

### PR DESCRIPTION
**Pull request by gniezen**
This PR is to:
- support G5 receivers
- add links to Mac USB driver to solve El Capitan issue
- minor UI improvements (Try Again button and Error Details)

### CODE REVIEW

**Comment by gniezen** [comment]
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;qh1wprAf&#x2F;24-uploader-link-to-all-in-one-usb-drivers-in-uploader"> [uploader] link to all-in-one USB drivers in Uploader</a>

**Comment by gniezen** [comment]
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;QHyR9yE7&#x2F;9-1-ux-3-1-dev-1-tst-uploader-dexcom-bundle-of-improvements"> &lt;1:ux, 3+1:dev, 1:tst&gt; [uploader] Dexcom bundle of improvements</a>

**Comment by gniezen** [comment]
Ping @jebeck for review.

File: description.txt [hunk 1]
```diff
@@ -1,7 +1,7 @@
 No more wrestling with Java plug-in errors, computer incompatibilities, or user flows that leave you confused. And no more proprietary data silos. The Tidepool Uploader works with Mac and PC.
 
 ■ Supported devices
-Dexcom G4 Platinum, Insulet OmniPod, Medtronic Minimed (523 and up), Abbott Precision Xtra.
+Dexcom (G4 Platinum and G5), Insulet OmniPod, Medtronic Minimed (523 and up), Abbott Precision Xtra.
```
Comment (jebeck) [inline description.txt]: Want to touch base w/product about whether we want to update this further to include the other currently supported BG meters?

**Comment by jebeck** [comment]
Tried an upload w/my empty/never-used receiver (after setting up time etc., even set the transmitter ID but didn't start a sensor session or anything).

Still errored out, not sure if that's expected, but it also looks to me as if there could be a regression in the error messaging:
screenshot 2016-01-13 18 12 51

Will put the console log in Slack for you.

File: lib/components/DeviceSelection.jsx [hunk 2]
```diff
@@ -36,15 +36,35 @@ var DeviceSelection = React.createClass({
...
+      if((window.app.state._os === 'mac') && (upload.mac !== undefined) ) {
+        driverLink = upload.mac.driverLink;
+      }
+
+      if((window.app.state._os === 'win') && (upload.win !== undefined) ) {
+        driverLink = upload.win.driverLink;
+      }
+
+      var displayText = '';
+      if (isChecked && (driverLink !== '')) {
```
Comment (jebeck) [inline lib/components/DeviceSelection.jsx]: I'm going to channel @jh-bate here and ask you to use `!_.isEmpty(driverLink)` - that's kind of our standard.

File: lib/components/UploadList.jsx [hunk 3]
```diff
@@ -69,13 +69,19 @@ var UploadList = React.createClass({
     var showDetailsThisUpload = _.includes(this.state.showErrorDetails, upload.key);
     var errorDetails = showDetailsThisUpload ? (<div className="UploadList-error-details">{upload.error.debug}</div>) : null;
     var showErrorsText = showDetailsThisUpload ? this.props.text.HIDE_ERROR : this.props.text.SHOW_ERROR;
-    
+    var errorMessage = upload.error.driverLink ? <div className="UploadList-error-message-wrapper">
```
Comment (jebeck) [inline lib/components/UploadList.jsx]: Is there a way I can try out the `driverLink` thing or is that impossible on my system w/pre-El Capitan OS X?

**Comment by jebeck** [comment]
A couple of issues listed above I'd like addressed and I'm not sure if there's more checking I can do re: the system-dependent driver links, but otherwise LGTM.

File: lib/components/UploadList.jsx (hunk 3 above)
Comment (gniezen) [inline lib/components/UploadList.jsx]: To see the error message with link to driver, just try to upload without a Dexcom connected. The driver works on pre-El Capitan as well, so there's no harm in Yosemite users installing it. In fact, if you have Portrait on the Web installed, you should already have it installed as well. To check, do a `kextstat | grep Dex` while the receiver is plugged in. If it displays something like `com.dexcom.driver.DexComUSB`, you've got it.

Comment (jebeck) [inline lib/components/UploadList.jsx]: Mmkay, just tried w/no Dexcom attached to see the error. Is the detail message re: powering off and on what's expected? I think it is, but wanted to check w/you to be certain:

screenshot 2016-01-14 10 28 57

**Comment by jebeck** [comment]
Remember to add the underline back to Error details/Hide details before merge.

Also, it occurred to me it'd be convenient to fold the bootstrapping logging in here (so that it's easier to debug with affected users once it's on prod). OK with you if put up a PR against this branch to add that?

File: lib/components/UploadList.jsx (hunk 3 above)
Comment (gniezen) [inline lib/components/UploadList.jsx]: Yes, I wish there was a way of determining if you're running El Capitan or Yosemite, but at the moment it'll display that error message to all Mac users if your device is not plugged in.

**Comment by gniezen** [comment]
Sure, put the logging as a PR against this branch, I can review and merge tomorrow.

**Comment by gniezen** [comment]
Addressed review feedback, including underlining error details, merging BtUTC logging PR and removing description.txt. Merging.

**Pull request merged**